lib/module-entry.js

```diff
@@ -0,0 +1,237 @@
+ * recursively so a whole tree can be computed all at once, independent of the
+ * hashing functions. We also compute the 'sourceHash' value (i.e. the hash of
+ * just the package source files, excluding the node_modules dependencies) at
+ * the same time to make it easier to establish the 'hash' value (includes
+ * dependency sourceHash values) later.
+ * Note this is a class function, not an instance function.
+ */
+ModuleEntry.locate = function(caches, name, dir, hashTreeFn) {
+  var h = heimdall.start('hashForDep:ModuleEntry.locate');
+  var Constructor = this;
+
+  try {
+    var nameDirKey = cacheKey(name, dir);
+    var realPathKey;
+
+    // It's possible that for a given name/dir pair, there is no package.
+    // Record a null entry in CACHES.PATH anyway so we don't need to
+    // redo that search each time.
+    if (caches.PATH.has(nameDirKey)) {
+      realPathKey = caches.PATH.get(nameDirKey);
+      if (realPathKey !== null) {
+        return caches.MODULE_ENTRY.get(realPathKey);
+      } else {
+        return null;
+      }
+    }
+
+    // There is no caches.PATH entry. Try to get a real package.json path.
+    // If there isn't a real path, the name+dir reference is
+    // invalid, so note that in caches.PATH. If there is a real path,
+    // check if it is already in the caches.MODULE_ENTRY. If not, create it
+    // and insert it.
+    var realPath = resolvePackagePath(caches.REAL_FILE_PATH, name, dir);
+
+    console.log("### realPath resolved to " + realPath);
+
+    if (realPath === null) {
+      caches.PATH.set(nameDirKey, null);
+      return null;
+    }
+
+    // We have a path to a file that supposedly is package.json.
+    // We need to be sure that either we already have a caches.MODULE_ENTRY
+    // entry (in which case we can just create a caches.PATH entry to
+    // point to it) or that we can read the package.json file, at which
+    // point we can create a caches.MODULE_ENTRY entry, then finally the new
+    // caches.PATH ENTRY.
+
+    // Generate the cache key for a given real file path. This key is then
+    // used as the key for entries in the CacheGroup.MODULE_ENTRY cache
+    // and values in the CacheGroup.PATH cache.
+    realPathKey = cacheKey('hashed:' + realPath, '');
+
+    if (caches.MODULE_ENTRY.has(realPathKey)) {
+      caches.PATH.set(nameDirKey, realPathKey);
+      return caches.MODULE_ENTRY.get(realPathKey);
+    }
+
+    var thePackage = tryRequire(realPath);
+
+    console.log("package resolved to " + thePackage);
+
+    // if the package was not found, do as above to create a caches.PATH entry
+    // that refers to no path, so we don't waste time doing it again later.
+    if (thePackage === null) {
+      caches.PATH.set(nameDirKey, null);
+      return thePackage;
+    }
+
+    // We have the package object and the relevant keys.
+    var rootDir = realPath.slice(0, realPath.length - 12 /* index of `/package.json` */);
+
+    console.log("realPath rootDir = " + rootDir);
```

Comment:
I suspect this was leftover from earlier debugging, I believe it should be removed.

Reply:
Yes. Just in to help with Windows debugging and will be removed.